docs/blog/2018-06-08-life-after-layouts/index.md

```diff
@@ -0,0 +1,118 @@
+import React from "react"
+
+export default () => <h1>I’m in a layout?</h1>
+```
+
+Once we start the app, we’ll see the following in the browser console:
+
+![Screenshot of the generated markup with the layout wrapper around the page content.](simple-layout.png)
+
+Like magic, our page is wrapped with the `.app-wrapper` container.
+
+If you'd like to try out this app, check out the [`v1-layout` branch of the demo repo](https://github.com/jlengstorf/life-after-layouts/tree/v1-layout).
+
+### How it works in Gatsby V2
+
+In V2, layouts are no longer automatically applied to our pages (more on _why_ this decision was made in the next section).
+
+This ultimately boils down to two breaking changes and one recommendation:
+
+1.  **BREAKING CHANGE:** Components at `src/layouts/` are no longer automagically wrapped around page templates.
+2.  **BREAKING CHANGE:** The `children` prop in our layout components is no longer a function (unless you explicitly provide a function).
+3.  We now recommend moving your layout components alongside the rest of your components (e.g. into `src/components/`).
+
+If we upgrade our simplified app by running `yarn add gatsby@next react react-dom` (see [the V1 => V2 migration guide](#tktk-link-to-migration-guide) for more information on why we need to install React here), we need to upgrade our layout by moving `src/layouts/index.js` to `src/components/layout.js` and changing `children` from a function to a regular prop:
+
+```jsx
+import React from "react"
+
+export default ({ children }) => <div className="app-wrapper">{children}</div>
+```
+
+Next, in `src/pages/index.js` we need to explicitly include the `Layout` component and wrap our page in it:
+
+```jsx
+import React from "react"
+import Layout from "../components/layout"
+
+export default () => (
+  <Layout>
+    <h1>I’m in a layout!</h1>
+  </Layout>
+)
+```
+
+Once we've made these changes, we can run `yarn develop` and see the updated layout in our browser:
+
+![Screenshot of the updated layout in the browser.](simple-layout-v2.png)
+
+Check out the [`v2-layout` branch of the demo repo](https://github.com/jlengstorf/life-after-layouts) to see this in action, or see [the diff](https://github.com/jlengstorf/life-after-layouts/compare/v1-layout...v2-layout) for details on exactly what changed.
+
+## Why change the layouts?
+
+Looking at the difference in code, it might be tempting to think, "Doesn't that just create _more work_ for developers who want to use layouts?"
+
+In short, the decision to remove layouts was part of an effort to reduce _mystery issues_ in Gatsby projects.
+
+### What are mystery issues?
+
+In V1, if something breaks on your page, you'd likely go to the page, look through the code, and try to spot the bug. But what if the bug is in the magic layout component? That component isn't referenced anywhere on the page, and if you don't _already know_ that Gatsby V1 wraps pages with layouts, it's pretty difficult to figure out what's going on.
```

Comment:
The main problem the layout component caused was it proved very difficult to share data between pages and the layouts e.g. people would want to pass data from the page to the layout to modify how the layout component would render and found that really hard. That and the reverse — passing layout to the page — was super hard. E.g. Gatsbygram got way simpler in v2. By breaking the normal React compositional model, passing data around got really hard / impossible.

Reply:
Good call. Adding a section above this one to address that.